Redazione Decimo Sprint (#225)
SevenBitsSwe/7BitsDocs: PB/Documentazione Esterna/Piano_di_Progetto.tex

        Di questa cifra, 1.879,44€ rappresentano i costi ancora da sostenere per completare il prodotto e la relativa documentazione.\\
        Questo trend positivo indica una gestione efficiente delle risorse e un controllo accurato dei costi. Il gruppo può quindi procedere con maggiore sicurezza verso le fasi finali del progetto, mantenendo comunque alta l'attenzione sulla qualità del lavoro e sul rispetto delle scadenze.

        %%%%%%%%%%
        %%%%% Decimo sprint
        %%%%%%%%%%%
    
        \newpage
        \subsubsection{Decimo sprint}
        \label{decimo-sprint$_G$}
    
        \paragraph{Durata sprint}\mbox{}\\
            \vspace{-1.5em}
            \begin{table}[h] 
            \renewcommand{\arraystretch}{1.2}  
            \begin{tabular}{ l l }
                Inizio: & 2025-03-24 \\
                Fine Prevista: & 2025-03-30 \\
                Fine Effettiva: & 2025-03-30 \\
                Giorni di ritardo: & 0 \\
            \end{tabular}
            \end{table}
            \vspace{-2em}
            {\renewcommand{\arraystretch}{1.5}%
            
            \paragraph{Panoramica generale e obiettivi}\mbox{}\vspace{0.4em}

>             Nel corso del decimo sprint, il gruppo si propone di terminare la codifica del prodotto MVP, inclusi i testi di integrazione e di Sistema. Inoltre dopo aver progettato il diagramma della classi nella sua versione definitiva, l'obiettivo sarà terminare i documenti di Specifica\_Tecnica e Manuale\_Utente per poi preparare la presentazione necessaria alla revisione \textit{PB}.

Reviewer: ```suggestion
            Nel corso del decimo sprint, il gruppo si propone di terminare la codifica del prodotto MVP, inclusi i testi di integrazione e di sistema. Inoltre, dopo aver progettato il diagramma della classi nella sua versione definitiva, l'obiettivo sarà terminare i documenti di \textit{Specifica\_Tecnica} e \textit{Manuale\_Utente} per poi preparare la presentazione necessaria alla revisione PB.
```

            Il gruppo ha in programma di:
            \begin{itemize}

>                 \item Terminare lo sviluppo dell' MVP;

Reviewer: ```suggestion
                \item Terminare lo sviluppo dell'MVP;
```

>                 \item Terminare la scrittura dei documenti Specifica\_Tecnica e Manuale\_Utente

Reviewer: ```suggestion
                \item Terminare la scrittura dei documenti \textit{Specifica\_Tecnica} e \textit{Manuale\_Utente};
```

                \item Proseguire con la rifinitura degli altri documenti visto l'avvicinamento della data di consegna.
            \end{itemize}
            Segue l'elenco delle attività inserite nel backlog per questa iterazione:
            \vspace{-0.5em}
            \begin{itemize}
            \setlength\itemsep{-0.2em}
                \item [-] Verifica Finale Codice Repository;
                \item [-] Verifica rispetto Standard PEP8;

>                 \item [-] Verifica Finale Documento Specifica\_Tecnica;

Reviewer: ```suggestion
                \item [-] Verifica Finale Documento \textit{Specifica\_Tecnica};
```

>                 \item [-] Verifica Finale Documento Manuale\_Utente;

Reviewer: ```suggestion
                \item [-] Verifica Finale Documento \textit{Manuale\_Utente};
```

>                 \item [-] Redazione Presentazione per Product Baseline, revisione Cardin;

Reviewer: ```suggestion
                \item [-] Redazione Presentazione per Product Baseline, revisione Cardin.
```

            \end{itemize}
    
            \paragraph{Gestione Rischi}\mbox{}
            \vspace{-1em}
            \subparagraph*{Rischi Attesi}\mbox{}
            

>             Relativamente alla nona iterazione, il gruppo ha identificato un potenziale rischio legato ad eventuali cambiamenti improvvisi a ridosso della consegna della \textit{Prodcut Baseline}. Questo a causa di errori che richiedono dei cambiamenti e una revisione delle attività gia svolte oppure dovuto a incomprensioni tra i membri del team su alcune task.

Reviewer: ```suggestion
            Relativamente alla decima iterazione, il gruppo ha identificato un potenziale rischio legato ad eventuali cambiamenti improvvisi a ridosso della consegna della Product Baseline. Questo a causa di errori che richiedono dei cambiamenti e una revisione delle attività già svolte oppure dovuto a incomprensioni tra i membri del team su alcune task.
```

                    \pie[
                        text=pin,
                        hide number,
                        color={responsabile, verificatore, progettista, programmatore, amministratore, analista}, 
                        radius=1.5
                    ]{5/Responsabile - 5\%, 9/Amministratore - 9\%, 10/Analista - 10\%, 25/Verificatore - 25\%, 23/Progettista - 23\%, 28/Programmatore - 28\%}
                \end{tikzpicture}
                \caption{Diagramma circolare della partizione delle ore consuntive per ruolo - Decimo sprint}
                \label{fig:Diagramma circolare della partizione delle ore consuntive per ruolo - Decimo sprint}
            \end{figure}
    
            \paragraph{Rendicontazione delle risorse economiche restanti}\mbox{}\vspace{0.4em}
            \begin{figure}[H]
                \centering
                \includegraphics[width=0.6\linewidth]{oreCostiDecimoSprint.png}
                \caption{Consuntivo orario e costi per ruolo - Decimo sprint}
                \label{fig:Consuntivo orario e costi per ruolo - Decimo sprint}
            \end{figure}
            
            \paragraph{Retrospettiva}\mbox{}\vspace{0.4em}

>             Durante il decimo sprint, il gruppo ha notato la fondamentale importanza di adottare una chiara comunicazione tra membri del team, soprattuto duranto lo sviluppo del core del software. Essendo una piccola modifica, non ha intaccato la data di consegna, ma potenzialmente avrebbe potuto creare problemi di rilevanza maggiore. Sono stati inoltre terminati i documenti necessari alla revisione PB con il Professor. Cardin, ed è stata iniziata la redazione della presentazione necessaria al colloquio.

Reviewer: ```suggestion
            Durante il decimo sprint, il gruppo ha notato la fondamentale importanza di adottare una chiara comunicazione tra membri del team, soprattuto duranto lo sviluppo del core del software. Trattandosi di una piccola modifica, non ha intaccato la data di consegna, ma potenzialmente avrebbe potuto creare problemi di rilevanza maggiore. Sono stati inoltre terminati i documenti necessari alla revisione PB con il Professor. Cardin, ed è stata iniziata la redazione della presentazione necessaria al colloquio.
```

    
            \subparagraph{Analisi delle strategie di mitigazione adottate}\mbox{}\vspace{0.4em}
            

>             Per mitigare il problema sorto, è stato necessario organizzare una piccola sessione di \textit{peer-to-peer progamming} dove i programmatori hanno in poche ore, hanno potuto chiarire e risolvere il problema prensente nella business logic senza troppe difficoltà. Questo fa capire che lavorare in maniera sincrona può risultare più costoso in termini di tempo e risorse, ma permette una maggiore produttività, poiché facilita la comunicazione immediata, la risoluzione rapida dei problemi e una più stretta collaborazione tra i membri del team.

Reviewer: ```suggestion
            Per mitigare il problema sorto, è stato necessario organizzare una piccola sessione di \textit{peer-to-peer progamming} durante la quale i programmatori hanno potuto chiarire e risolvere il problema presente nella business logic senza troppe difficoltà. Questo ha evidenziato come il lavoro sincrono, sebbene più oneroso in termini di tempo e risorse, favorisca una maggiore produttività, in quanto facilita la comunicazione immediata, accelera la risoluzione dei problemi e promuove una collaborazione più stretta tra i membri del team.
```
